#!/usr/bin/env python
PACKAGE = "dynamic_tutorials"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("alpha",    double_t,    0, "Alpha", 50,  0, 100)
gen.add("dist_dubin_shot",    double_t,    0, "Min_dist_from_obst", 8,  0, 100)
gen.add("max_obs_dist", double_t, 0, "Maximum Obstacle distance",    50, 0,   100)
gen.add("carlx", int_t, 0, "Car upper-left x",    -5, -100,   100)
gen.add("carly", int_t, 0, "Car upper-left y",    -5, -100,   100)
gen.add("carrx", int_t, 0, "Car bottom-right x",    5, -100,   100)
gen.add("carry", int_t, 0, "Car bottom-right y",    5, -100,   100)
gen.add("planner_grid_resolution", double_t, 0, "Planner Reslution ",    0.5, 0, 2)
gen.add("WT",    double_t,    0, "Voronoi Weight", 0.3,  0, 5)
gen.add("DISPLAY_PATH",    bool_t,    0, "Display Path", False)
gen.add("DISPLAY_SEARCH_TREE",    bool_t,    0, "Display Search Tree", False)
gen.add("BOT_W", double_t, 0, "Bot width",    1.9, 1,   5)
gen.add("BOT_L", double_t, 0, "Bot length",    2.8, 1,   5)
gen.add("BOT_MAX_ALPHA", int_t, 0, "Bot max alpha",    45, -90,   90)
gen.add("dist_replan",    double_t,    0, "dist_replan", 2,  0, 100)


exit(gen.generate(PACKAGE, "hybrid_astar", "params_hybrid_astar"))
